sdk/kubeflow/trainer/constants/constants.py

```diff
@@ -90,40 +86,95 @@
 
+# The name of the ReplicatedJob and container of the dataset initializer.
+# Also, it represents the `trainjob-ancestor-step` label value for the dataset initializer step.
+DATASET_INITIALIZER = "dataset-initializer"
+
+# The name of the ReplicatedJob and container of the model initializer.
+# Also, it represents the `trainjob-ancestor-step` label value for the model initializer step.
+MODEL_INITIALIZER = "model-initializer"
+
 # The default path to the users' workspace.
 # TODO (andreyvelich): Discuss how to keep this path is sync with pkg.initializers.constants
 WORKSPACE_PATH = "/workspace"
 
-# The name of the ReplicatedJob and container of the dataset initializer
-DATASET_INITIALIZER = "dataset-initializer"
-
 # The path where initializer downloads dataset.
 DATASET_PATH = os.path.join(WORKSPACE_PATH, "dataset")
 
-# The name of the ReplicatedJob and container of the model initializer
-MODEL_INITIALIZER = "model-initializer"
-
 # The path where initializer downloads model.
 MODEL_PATH = os.path.join(WORKSPACE_PATH, "model")
 
-# The Job name for the launcher (e.g. mpirun launcher).
-JOB_LAUNCHER = "launcher"
-
-# The container name for the launcher
-CONTAINER_LAUNCHER = "launcher"
-
-# The Job name for the trainer nodes.
-JOB_TRAINER_NODE = "trainer-node"
+# The name of the ReplicatedJob and container of the node. The node usually represents single
+# VM where distributed training code is executed.
+# TODO: Change it to "node"
+NODE = "trainer-node"
 
-# The Pod type for the master node.
-MASTER_NODE = f"{JOB_TRAINER_NODE}-0"
+# The `trainjob-ancestor-step` label value for the trainer step.
+TRAINER = "trainer"
 
 # The Pod pending phase indicates that Pod has been accepted by the Kubernetes cluster,
 # but one or more of the containers has not been made ready to run.
 POD_PENDING = "Pending"
 
-# The container name for the Trainer.
-CONTAINER_TRAINER = "trainer"
+# The default PIP index URL to download Python packages.
+DEFAULT_PIP_INDEX_URL = os.getenv("DEFAULT_PIP_INDEX_URL", "https://pypi.org/simple")
+
+# The default command for the Trainer.
+DEFAULT_COMMAND = ["bash", "-c"]
 
 # The Torch env name for the number of procs per node (e.g. number of GPUs per Pod).
 TORCH_ENV_NUM_PROC_PER_NODE = "PET_NPROC_PER_NODE"
+
+# The container entrypoint for distributed PyTorch.
+TORCH_ENTRYPOINT = "torchrun"
+
+# The name of the ReplicatedJob to launch mpirun.
+MPI_LAUNCHER = "launcher"
+
+# The OpenMPI env name for the number of slots per nude (e.g. number of GPUs per Pod).
+MPI_ENV_NUM_SLOTS_PER_NODE = "OMPI_MCA_orte_set_default_slots"
+
+# The container entrypoint for distributed MPI
+MPI_ENTRYPOINT = "mpirun"
+
+
+# The dict where key is the container image and value its representation.
+# Each Trainer representation defines trainer parameters (e.g. type, framework, entrypoint).
+# TODO (andreyvelich): We should allow user to overrides the default image names.
+ALL_TRAINERS: Dict[str, types.Trainer] = {
```

Comment:
@astefanutti @kubeflow/wg-training-leads @saileshd1402 @Electronic-Waste Please let me know what do you think about this model to detect Runtime Trainers from the image name?